autobahn/websocket/interfaces.py

```diff
@@ -36,6 +36,32 @@
            'IWebSocketChannelStreamingApi')
 
 
+@six.add_metaclass(abc.ABCMeta)
+class IWebSocketClientAgent(object):
+    """
+    Instances implementing this interface create WebSocket
+    connections.
+    """
+
+    def open(self, transport_config, options, protocol_class=None):
```

Comment:
I guess `transport_config` is again for compatibility with existing autobahn APIs... a real type here instead of a dict would be pretty nice though.